### ISSUE TITLE
headers was strtolower. X-Robots-Tag became x-robots-tag

### ISSUE THREAD

**Issue by itbdw** [symfony/symfony#23698]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 2.6.0

I want set a custom header with `X-Robots-Tag` in some response cases. But it turn to a lower case, became `x-robots-tag`. What should a do?

Just hard code for now... `vendor/symfony/http-foundation/ResponseHeaderBag.php`
```
     /**
     * {@inheritdoc}
     */
    public function set($key, $values, $replace = true)
    {
        ...

        if ($uniqueKey == 'x-robots-tag') {
            $this->headerNames['x-robots-tag'] = 'X-Robots-Tag';
        }
    }
```

**Comment by rskuipers** [comment]
@itbdw The headers are only stored with strtolower, when it's being outputted it is converted to uppercase as expected (note the ucfirst in the code below):

```php
    /**
     * Returns the headers as a string.
     *
     * @return string The headers
     */
    public function __toString()
    {
        if (!$this->headers) {
            return '';
        }

        $max = max(array_map('strlen', array_keys($this->headers))) + 1;
        $content = '';
        ksort($this->headers);
        foreach ($this->headers as $name => $values) {
            $name = implode('-', array_map('ucfirst', explode('-', $name)));
            foreach ($values as $value) {
                $content .= sprintf("%-{$max}s %s\r\n", $name.':', $value);
            }
        }

        return $content;
    }
```

**Comment by javiereguiluz** [comment]
According to [the relevant section of the related standard](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2), the name of the HTTP headers are case insensitive, both request and response.

**Comment by itbdw** [comment]
Thanks @rskuipers , I encountered this problem where I use `abort()` in Laravel. And As a result, I found this maybe a Laravel bug, It just pass the `$headers->all()` instead of `$headers->allPreserveCase()`, and I made a pull request, waiting...

Thanks @javiereguiluz , >the name of the HTTP headers are case insensitive What I wanna make clear is I just wanna `Symfony` keep the original capitalizations. And it does. I think it's a bug of Laravel.
